content/docs/user-guide/programming/components/reflection/behavior-context.md

```diff
@@ -0,0 +1,461 @@
+---
+
+The behavior context makes runtime code accessible to O3DE scripting systems such as Script Canvas or Lua. It provides script bindings that invoke runtime C++ methods, read constants, write properties, and create and handle EBus events. You can have multiple behavior contexts that are specialized for different purposes, and you can unreflect the behavior contexts to implement reloading.
+
+Use the behavior context to bind the following C++ constructs for scripting:
+
++ [Classes](#classes)
++ [Methods](#methods)
++ [Properties](#properties)
++ [Constants](#constants)
++ [Enums](#enums)
+
+In addition, the behavior context supports the O3DE [EBus](/docs/user-guide/programming/ebus) and [AZ::Event](/docs/user-guide/programming/az-event) event systems:
+
++ [EBus](#ebus)
+  + [Events](#ebus-events)
+  + [Event Handlers](#ebus-event-handlers)
+
+## Classes
+
+Reflects a C++ class or struct. You can provide an optional name. If you do not provide a name, the class name from `AzTypeInfo` is used. That name must be unique for the scope. Because the system uses `AzRTTI` to build the class hierarchy, you can use RTTI if you want to reflect base class functionality.
+
+Classes that are bound to the behavior context become objects that can be instantiated in a script environment. To reflect a class, you must provide the type that is reflected as a template argument to the class function.
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorcontext->Class<MyClass>();
+}
+```
+
+The base class should also be specified, if applicable:
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorcontext->Class<MyClass, TheBaseClass>();
+}
+```
+
+### Additional class information
+
++ **Allocator** - You can provide a custom allocator/ deallocator for your class. This allows you to override any existing allocation schema. If you do not provide a custom allocator, `aznew` is used (`AZ_CLASS_ALLOCATOR`).
++ **Constructor** - Allows you to enumerate the class constructors that you want to reflect. You must pass all constructor arguments as template arguments.
++ **Wrapping/WrappingMember** - Allows code to inform the system that it is a wrapper of another class. This is useful when you reflect smart pointers and string wrappers.
++ **Userdata** - Allows you to provide a pointer to user data. The pointer is accessible from all callbacks (like a custom allocator) that you implement for the class.
++ **Method** - Reflects a C++ class function. The first argument is the class pointer. This is the same usage as global methods.
++ **Property** - Reflects class data. The first argument is the class pointer. This is the same usage as global properties.
++ **Enum** - Enums are read-only `int` properties.
++ **Constant** - Constants are read-only properties.
+
+### Attributes
+
+You can use the following, optional built-in attributes to decorate a class.
```

Comment:
Suggest including an example showing how to set a variety of attributes during the context registration - seems to be a common pattern used in O3DE, especially for the editor context.

Reply:
Definitely. Will do...

Reply:
Added.